server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpDecodeInput.scala

```diff
@@ -0,0 +1,37 @@
+package sttp.tapir.server.stub
+
+import sttp.client.Request
+import sttp.model.Method
+import sttp.model.Uri.QuerySegment
+import sttp.tapir.model.ServerRequest
+import sttp.tapir.server.internal.DecodeInputsContext
+
+class SttpDecodeInput(r: Request[_, _], segmentIndex: Int = 0) extends DecodeInputsContext {
+  override def method: Method = Method(r.method.method)
+
+  override def nextPathSegment: (Option[String], DecodeInputsContext) =
+    (r.uri.pathSegments.drop(segmentIndex).headOption.map(_.v), new SttpDecodeInput(r, segmentIndex + 1))
+
+  override def header(name: String): List[String] = r.headers.filter(_.name.toUpperCase == name.toUpperCase).map(_.value).toList
+
+  override def headers: Seq[(String, String)] = r.headers.map(h => (h.name, h.value))
+
+  override def queryParameter(name: String): Seq[String] =
+    r.uri.querySegments
+      .collect {
+        case qp @ QuerySegment.KeyValue(k, _, _, _) if k == name => qp
+      }
+      .map(_.v)
+
+  override def queryParameters: Map[String, Seq[String]] =
+    r.uri.querySegments
+      .collect {
+        case qp @ QuerySegment.KeyValue(_, _, _, _) => qp
+      }
+      .groupBy(_.k)
+      .map { case (k, vs) => (k, vs.map(_.v)) }
+
+  override def bodyStream: Any = ()
```

Comment:
here we could "do our best" and see if the given request's body is a `StreamBody`?

Reply:
I don't know how streaming works in Tapir yet, let me check what can be done here when I finish dealing with input/output decoding/encoding :)

Reply:
same as in sttp, passes the (unchanged) stream to the backend